### PR TITLE
Read files from local fileshare and upload to Azure Data Lake

### LINKED ISSUE #84: Upload files into Azure Data Lake
1. Have a folder with 3 example data files
2. Upload the files from the folder to Azure Data Lake

Parameters:
1. Folder path with files to be uploaded
2. Azure Data Lake target container
3. Azure Data Lake target path inside the container (created the folder if it does not exist, see upload task if it has this option already)
5. Azure Data Lake credentials

### PR DESCRIPTION
**Pull request by jannejjj**
closes #84 

## Summary by CodeRabbit

- **New Features**
  - Introduced a new process to upload files from a local filesystem to Azure Data Lake, including prerequisites, implementation details, and error handling.
  - Added metadata support for the new file upload process.
  - Enabled overwrite functionality by default when transferring files to Azure Data Lake.

### CODE REVIEW

File: Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/long-description.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+This Process uploads files from a **local filesystem** based on a **file mask** and a **source path** to **Azure Data Lake**.
+
+![Template](assets/Upload_files_from_local_fileshare_to_Azure_Data_Lake.svg)
+
+# Prerequisites
+
+This template assumes that the following prerequisites are in place:
+
+- The Frends agent has access to the local files that need to be uploaded.
+- The Azure user has the permission to upload files to the Data Lake.
```
Comment (FrendsSarlinS) [inline Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/long-description.md:10]: Should "the permission" be "rights" instead in this context?

**Comment by FrendsSarlinS** [comment]
I had a look at the template and commented on one thing in the markdown file, but nothing much to comment on the process itself. It's simple and functional!

File: Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/assets/Upload_files_from_local_fileshare_to_Azure_Data_Lake.svg
Comment (jefim) [inline Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/assets/Upload_files_from_local_fileshare_to_Azure_Data_Lake.svg:1]: 1. Lets keep the more usual "downwards" style and move the "else" branch below the if. This style is used in 99.9% tempaltes and generally in the processes we do. 
2. In the throw shape - change the mode to Text and remove those quotes, would just look nicer as we do not need an expression there.
3. In the throw shape - include the ErrorMessage property from UploadFiles task (keep the Text mode and include using the {{#result...}} syntax.
4. In the return shape also change to Text mode and include the variables using the {{...}} syntax.

File: Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/long-description.md (hunk 1 above)
Comment (jefim) [reply Templates/Files/Local Files/Upload files from local fileshare to Azure Data Lake/long-description.md:10]: Whatever word you choose - it should be plural. So either permissions or rights.

**Comment by jannejjj** [comment]
Thanks for the comments, addressed the given points. 😄

**Pull request merged**